### ISSUE TITLE
embedded connect CCP is not working with the streams api for Chrome

### ISSUE THREAD

**Issue by Demireon** [amazon-connect/amazon-connect-streams#112]
I am having recent problems with two applications that use an embedded Connect CCP and the Amazon-Connect-Streams API in that the ringtone functionality is not working in Chrome for agents. When an agent receives a call, there is a notification but no ringing. This functionality works as normal and expected using Firefox, however. 

I saw that this was an issue previously (https://forums.aws.amazon.com/ann.jspa?annID=5400) with chrome version update 64, and the recommended guidance was to build and deploy the latest version of this API. I have followed those instructions but am still observing these problems. 

Have there been any problems with this api functionality with the recent version 75 update to Chrome?

**Comment by mschersten** [comment]
Are these production applications that it just stopped working, or are you in development?  Chrome has a feature that prevents a page from playing audio before a user interacts with the page, so I found that during development I would refresh the page, load as available, and the ring from my test call was before I clicked on anything.  I solved this by setting the agents to offline if they loaded as available so that they would have to click on something before getting a call.

**Comment by Demireon** [comment]
So far I have only tested in the development environment, but yeah I saw that that was a feature for chrome as well. I have followed those directions of interacting with the UI, and I keep getting the same problem. 

I temporarily downgraded to Chrome version 71, and my applications work correctly in those environments. I think a recent update to Chrome has brought back these issues with the streams api. Also, I have ensured that all of the microphone/sounds permissions are set properly to allow the ringtone to work. 

Has the recent version 75 upgrade to Chrome caused any problems on your end with apps that utilize this API?

**Comment by jeffaxial** [comment]
I am having the same issue. Audio just doesn't seem to work across the board for me on Chrome 75. My team didn't seem to have any issues previously that I am aware of.

**Comment by jeffaxial** [comment]
@Demireon Do hide the CCP and have your own UI?

I have a custom UI built on top of the streams API and we're hiding the CCP. But I'm now realizing showing the CCP instantly makes audio work. I have to assume Chrome is suppressing hidden audio sources (which as a user is great...as a developer using Connect, not so much).

**Comment by Demireon** [comment]
@jeffaxial - Seems like the problem is indeed about Chrome suppressing certain audio sources then. We're actually presenting the CCP in our UI, and even interacting with it before doing a call. the audio is still not being played for the user, but interestingly the dialpad beep sounds are always working.

**Comment by hfahlbusch** [comment]
I just created a pull request for issue #81 - this addresses autoplay error in chrome. May you could check if this fixes your problem too.

**Comment by Demireon** [comment]
Hi Helge! I tried out the code in your PR, but unfortunately I was not able to fix the problem I was having with the ringer not working with the recent version of Chrome.

**Comment by hfahlbusch** [comment]
Hi Demireon,

I am using it with the latest chrome version (Version 75.0.3770.100 (64-Bit)) on mac and it works.
So I guess there something special in your code or in the chrome settings.

Is it showing any errors in chrome console? 
Are the domains (your ui domain / ccp domain) allowed to play sound?
Is it directly embedded or do you have some redirection or multiple nested iframes?

**Comment by prosunshining** [comment]
I could make it working with @hfahlbusch 's PR(`iframe.allow = "microphone; autoplay";`).
CCP Container should have style of `visibility: true`, I used to hide embedded ccp container with `visiblity: hidden` and that caused an issue playing audio in iframe.

**Comment by camstuart** [comment]
Similar challenges here;

When I use various tricks to hide the embedded ccp (`display: none;` or zero div height and width) I get no audio (but microphone seems ok) such as ringing sound, or the caller being able to hear me.

*Audio Working*, but ccp visible:
```html
<div id="ccpContainerDiv" style="width: 320px; min-width: 200px; height: 465px; min-height: 400px;">
    <!--Amazon CCP is hiding in here-->
  </div>
```

*Audio Not Working*, but ccp hidden, as I want it to be:
```html
<div id="ccpContainerDiv" style="width: 320px; min-width: 200px; height: 465px; min-height: 400px; display: none;">
    <!--Amazon CCP is showing on the page, undesirable -->
  </div>
```

**Comment by rswift** [comment]
I'm about to embark on embedding CCP just like this, so wondering, @camstuart does a 1x1 div work?

**Comment by camstuart** [comment]
Well done @rswift yes, that actually worked. My final div setup:

```html
<div id="ccpContainerDiv" style="width: 1px; height: 1px;">
    <!--Amazon CCP is hiding in here-->
</div>
```

**Comment by jeffaxial** [comment]
`height: 0; overflow: hidden;`

...also does the trick. It's a shame we have to even interface with the ccp container to just utilize the API.

**Comment by prosunshining** [comment]
I could solve the issue with the latest chrome by making embedded ccp visible but placing outside of window that is available, and applying @hfahlbusch 's PR #81 .

1) No display:none, visibility:hidden for ccp container div.
2) iframe.allow = "microphone; autoplay;"; for connect stream api.

With those changes, custom ccp plays ringtone when call comes in.

**Comment by hfahlbusch** [comment]
I guess there has to be at least some "visible" surface for chrome to play audio.
For videos there is a min size in some parts.
Maybe we could add a feature or css class that brings needed stylings.
